File: demo/src/main/java/com/example/demo/entities/Employee.java
package com.example.demo.entities;

import java.time.LocalDate;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.example.demo.enums.Gender;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Past;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Table(name = "employees")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    @NotBlank(message = "Họ tên không được để trống")
    @Size(min = 2, max = 100)
    private String fullName;

    @Column(nullable = false, unique = true, length = 100)
    @NotBlank(message = "Email không được để trống")
    @Size(max = 100)
    private String email;

    @Column(nullable = false)
    @NotNull(message = "Ngày sinh không được để trống")
    @Past
    private LocalDate dateOfBirth;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    @NotNull(message = "Giới tính không được để trống")
    private Gender gender;

    @Column(nullable = false)
    @NotBlank(message = "Số điện thoại không được để trống")
    @Pattern(regexp = "^\\d{10}$", message = "Số điện thoại phải gồm 10 chữ số")
    private String phoneNumber;

    @Column(nullable = false)
    @NotNull(message = "Trạng thái không được để trống")
    private Boolean active = true;

    @Column(nullable = false)
    @NotBlank(message = "Mật khẩu không được để trống")
    private String hashedPassword;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    // Getters and Setters
}

File: my-app/src/types/employee.ts

export type Gender = 'MALE' | 'FEMALE' | 'OTHER';


export interface Employee {
  id: number;
  fullName: string;
  email: string;
  dateOfBirth: string; // ISO string
  gender: Gender;
  phoneNumber: string;
  active: boolean;
  createdAt: string;
  updatedAt: string;
}


export interface EmployeeCreateRequest {
  fullName: string;
  email: string;
  dateOfBirth: string;
  gender: Gender;
  phoneNumber: string;
  password: string;
  active: boolean;
}


export interface EmployeeUpdateRequest {
  fullName?: string;
  email?: string;
  dateOfBirth?: string;
  gender?: Gender;
  phoneNumber?: string;
  password?: string;
  active?: boolean;
}

export interface ErrorResponse {
  message: string;
  status: number;
  timestamp: string;
}
